### PR TITLE
Dotnet branch distance

### PR DESCRIPTION
**Pull request by amidgol**

### CODE REVIEW

File: client-dotnet/src/EvoMaster.Instrumentation/Coverage/MethodReplacement/DistanceHelper.cs [hunk 1]
```diff
@@ -53,20 +82,26 @@ public class DistanceHelper {
...
 
-        Trace.Assert(dist >= 0);
+        if (dist < 0){
+            dist = long.MaxValue; // overflow
+        }
+
+        // Man: Amid, I commented this assert out.
+        // please check why you put such assert here
+        //Trace.Assert(dist >= 0);
         return dist;
```
Comment (ZhangMan1126) [inline client-dotnet/src/EvoMaster.Instrumentation/Coverage/MethodReplacement/DistanceHelper.cs:105]: could you check this comment? if you agree with it, then you could remove the comment

File: client-dotnet/src/EvoMaster.Instrumentation/Probes.cs [hunk 2]
```diff
@@ -52,5 +54,154 @@ public class Probes {
         public static void EnteringStatement(string className, int lineNo, int columnNo) {
             ExecutionTracer.EnteringStatement(className, lineNo, columnNo);
         }
+
+        public static void EnteringBranch(string className, int lineNo, int branchId) {
+            //TODO: remove
```
Comment (ZhangMan1126) [inline client-dotnet/src/EvoMaster.Instrumentation/Probes.cs:59]: might remove this as we discussed

**Review by ZhangMan1126 — CHANGES_REQUESTED** [review]
thanks, Amid! I leave few comments

File: client-dotnet/src/EvoMaster.Instrumentation/Coverage/MethodReplacement/DistanceHelper.cs (hunk 1 above)
Comment (amidgol) [reply client-dotnet/src/EvoMaster.Instrumentation/Coverage/MethodReplacement/DistanceHelper.cs:102]: removed

File: client-dotnet/src/EvoMaster.Instrumentation/Probes.cs (hunk 2 above)
Comment (amidgol) [reply client-dotnet/src/EvoMaster.Instrumentation/Probes.cs]: removed

File: ConsoleApp1/ConsoleApp1.csproj [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (arcuri82) [inline ConsoleApp1/ConsoleApp1.csproj:1]: what is this?

File: ConsoleApp1/Program.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (arcuri82) [inline ConsoleApp1/Program.cs:1]: what is this?

File: client-dotnet/src/EvoMaster.Instrumentation/Coverage/MethodReplacement/DistanceHelper.cs [hunk 5]
```diff
@@ -23,25 +24,54 @@ public class DistanceHelper {
...
+    public static double IncreasedDistance(double distance, double delta){
+        if (delta < 0){
+            throw new ArgumentException("Invalid negative delta: " + delta);
+        }
+
+        if (delta == 0){
+            throw new ArgumentException("Meaningless 0 delta");
+        }
+
+        if (double.IsFinite(distance) || distance == double.MaxValue){
```
Comment (arcuri82) [inline client-dotnet/src/EvoMaster.Instrumentation/Coverage/MethodReplacement/DistanceHelper.cs:36]: isn't this a bug, as always true?

File: client-dotnet/src/EvoMaster.Instrumentation/Coverage/MethodReplacement/DistanceHelper.cs [hunk 6]
```diff
@@ -23,25 +24,54 @@ public class DistanceHelper {
...
+        if (baseVale < 0 || baseVale >= 1){
+            throw new ArgumentException("Invalid base: " + baseVale);
+        }
 
-    public static int DistanceToDigit(char c) {
+        if (distance < 0){
+            throw new ArgumentException("Negative distance: " + distance);
+        }
+
+        if (double.IsFinite(distance) || distance == double.MaxValue){
```
Comment (arcuri82) [inline client-dotnet/src/EvoMaster.Instrumentation/Coverage/MethodReplacement/DistanceHelper.cs:62]: see previous comment. is this a bug?

File: client-dotnet/test/EvoMaster.Instrumentation.Tests/Coverage/MethodReplacement/DistanceHelperTest.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
+using System;
+using Xunit;
+
+namespace EvoMaster.Instrumentation.Tests.Coverage.MethodReplacement{
+    public class DistanceHelperTest{
+        
+        // Comment this out due to failing on CI
```
Comment (arcuri82) [inline client-dotnet/test/EvoMaster.Instrumentation.Tests/Coverage/MethodReplacement/DistanceHelperTest.cs:7]: test cases that for some reasons need to be disabled, should not be commented out, but rather "disabled".
in Java, can be done with `@Disabled`.
need to find equivalent in C#.
the reason is that they ll still show up when running all tests, although will be marked as skipped

Comment (arcuri82) [reply client-dotnet/test/EvoMaster.Instrumentation.Tests/Coverage/MethodReplacement/DistanceHelperTest.cs:7]: could it fail on CI due to different locales?

**Review by arcuri82 — CHANGES_REQUESTED** [review]

**Comment by arcuri82** [comment]
Hi,
 @Amid @ZhangMan1126 : what is state of this PR? still working on it? was there a branch as well in EMB that needs a PR?

**Comment by ZhangMan1126** [comment]
hi, @amidgol  will take over the comments which are not addressed, For EMB, we will create a RP for it.

File: ConsoleApp1/ConsoleApp1.csproj (hunk 3 above)
Comment (amidgol) [reply ConsoleApp1/ConsoleApp1.csproj]: It was a redundant project. Now it's removed

File: client-dotnet/test/EvoMaster.Instrumentation.Tests/Coverage/MethodReplacement/DistanceHelperTest.cs (hunk 7 above)
Comment (amidgol) [reply client-dotnet/test/EvoMaster.Instrumentation.Tests/Coverage/MethodReplacement/DistanceHelperTest.cs]: Yes, that's the case. Now it's skipped.

**Pull request merged**